### PR TITLE
Initialize the texture destruction callback in the Metal embedder test harness

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: shell/platform/embedder/tests/embedder_test_context_metal.cc [hunk 1]
```diff
@@ -100,6 +100,7 @@ FlutterMetalTexture EmbedderTestContextMetal::GetNextDrawable(
   texture.texture_id = texture_info.texture_id;
   texture.texture =
       reinterpret_cast<FlutterMetalTextureHandle>(texture_info.texture);
+  texture.destruction_callback = nullptr;
```
Comment (dnfield) [inline shell/platform/embedder/tests/embedder_test_context_metal.cc:103]: Should this get initialized to nullptr in the struct itself?

Comment (dnfield) [reply shell/platform/embedder/tests/embedder_test_context_metal.cc:103]: ...or can we not do that in a C struct

**Review by dnfield — APPROVED** [review]
LGTM

Verified locally this makes the ASAN failures in https://github.com/flutter/engine/pull/44813/ happy

File: shell/platform/embedder/tests/embedder_test_context_metal.cc (hunk 1 above)
Comment (jason-simmons) [reply shell/platform/embedder/tests/embedder_test_context_metal.cc:103]: no - the embedder API declarations are C compatible and do not offer a way to do this automatically.

Comment (chinmaygarde) [reply shell/platform/embedder/tests/embedder_test_context_metal.cc:103]: `FlutterMetalTexture texture = {};` should zero out a C struct. In this case we declare the struct on line 98.

Comment (jason-simmons) [reply shell/platform/embedder/tests/embedder_test_context_metal.cc]: Changed this to use `texture = {}`

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**